gitbook-docs/security.md

```diff
@@ -0,0 +1,118 @@
+# Security
+
+## Communications Security
+
+For privacy and data integrity REST and WebSockets communications should be secured with Transport Layer Security
+(TLS). All communications over unsecure protocols like HTTP and WebSockets without TLS must be considered insecure even
+with authentication and access control mechanisms in place.
+
+## Authentication
+
+Authentication for Signal K REST and WebSockets connections is based on HTTP cookies or tokens carried in the HTTP
+header.
+
+### Authentication via HTTP
+
+A device or a web client can authenticate with a Signal K server by providing a username and password via a standard
+HTTP POST request to `/signalk/«version»/auth/login`.
+
+The `«version»` field is the endpoint version identifier chosen by the client from those offered by the server. See the
```

Comment:
I replaced `v1` with `«version»`, but I wonder how authentication works when the server supports multiple versions. If a client logs in via `signalk/v1/auth/login` should they have access to `signalk/v2/api`?

Also, should the login path be included in the `endpoint` objects from `GET /signalk`?

Reply:
The login should  give a set of permissions (role) to the user. They may or may not include `/signalk/v2`. 
Lets also assume that `/signalk/v1/login` redirects to `/signalk/v2/login` if the server prefers that. Implementation detail really

Reply:
> Lets also assume that /signalk/v1/login redirects to /signalk/v2/login if the server prefers that. Implementation detail really

No, we can't do that. What if something changes between v1 and v2? The client might not be able to log in if it doesn't support v2.

Reply:
Probably an edge case for now, login is a well worn path. If the server advertised v2 and the client didnt support it then they are screwed anyway.

Reply:
Since 
- we don't know how future versions will work
- can not devise something decidedly future proof
- there is nothing obvious to be specified about this
I'd like to move forward without addressing this issue further.